### PR TITLE
It obtains the puppet master hostname from the puppet master url

### PR DESCRIPTION
**Pull request by hmunfru**
It obtains the puppet master hostname from the puppet master url obtained from keystone

### CODE REVIEW

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java [hunk 1]
```diff
@@ -222,12 +224,30 @@ public String getUserData(ClaudiaData claudiaData, TierInstance tierInstance) {
...
 
     }
 
+
+    /**
+     * It obtains the puppet master hostname, required for user data.
+     * @param puppetMasterUlr
+     * @return
+     */
+    public String getPuppetMasterHostname (String puppetMasterUlr){
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java]: puppetMasterUrl

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java [hunk 2]
```diff
@@ -222,12 +224,30 @@ public String getUserData(ClaudiaData claudiaData, TierInstance tierInstance) {
...
+                .replace("{if_up}", generateIfUp(tierInstance)).replace("{puppet_master}", puppetHostname);
         log.debug("payload " + file);
         return file;
 
     }
 
+
+    /**
+     * It obtains the puppet master hostname, required for user data.
+     * @param puppetMasterUlr
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java]: puppetMasterUrl

File: core/src/test/java/com/telefonica/euro_iaas/paasmanager/claudia/ClaudiaClientOpenStackImplTest.java [hunk 3]
```diff
@@ -313,9 +320,33 @@ public void testGenerateInterfacesFileTwoNet() throws Exception {
...
 
+    @Test
+     public void getPuppetMasterHostname() {
+        String hostname = claudiaClientOpenStack.getPuppetMasterHostname(HOSTNAME);
+        assertEquals(hostname, HOSTNAME);
+    }
+
+    @Test
+    public void getPuppetMasterHostname1() {
+        String hostname = claudiaClientOpenStack.getPuppetMasterHostname("http://"+ HOSTNAME);
```
Comment (jesuspg) [inline core/src/test/java/com/telefonica/euro_iaas/paasmanager/claudia/ClaudiaClientOpenStackImplTest.java]: rename test method: getPuppetMasterHostname1 by (e.g.) shouldGetPuppetMasterUrlWithoutPort

File: core/src/test/java/com/telefonica/euro_iaas/paasmanager/claudia/ClaudiaClientOpenStackImplTest.java [hunk 4]
```diff
@@ -313,9 +320,33 @@ public void testGenerateInterfacesFileTwoNet() throws Exception {
...
+        assertEquals(hostname, HOSTNAME);
+    }
+
+    @Test
+    public void getPuppetMasterHostname2() {
```
Comment (jesuspg) [inline core/src/test/java/com/telefonica/euro_iaas/paasmanager/claudia/ClaudiaClientOpenStackImplTest.java]: rename test method: getPuppetMasterHostname2 by shouldGetPuppetMasterUrlWithPort

**Comment by jesuspg** [comment]
question: it should be this "PR" a hotfix in master branch?

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java [hunk 5]
```diff
@@ -79,6 +79,8 @@
     private TierInstanceManager tierInstanceManager = null;
     private SystemPropertiesProvider systemPropertiesProvider = null;
     private static final int POLLING_INTERVAL = 10000;
+    private static String HTTP_START = "://";
+    private static String PORT_START = ":80";
```
Comment (jframos) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java]: If the port of the puppet-master/puppetdb service changes in the Keystone's service catalog... we will have to change the code. Is it possible to change the logic for getting the hostname using other pattern? Perhaps you could use a regular expression to do that.

**Comment by flopezag** [comment]
Coverage decreased (-0.15%)...

**Comment by hmunfru** [comment]
we have increase the coverage. https://coveralls.io/builds/2447663 45.45%
But it is not shown here

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java (hunk 5 above)
Comment (hmunfru) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java]: NTC

Comment (jframos) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java]: Are we going to improve it in this PR?

**Comment by flopezag** [comment]
The description of this issue make reference to some jira issue? In the same time you should specify which you think that should check the PR together with where do you check it?

**Comment by hmunfru** [comment]
It is related to https://jira.tid.es/browse/CLAUDIA-4940

**Comment by jframos** [comment]
LGTM.
Jira Issue: CLAUDIA-4940

**Comment by flopezag** [comment]
LGTM

**Comment by jesuspg** [comment]
LGTM

**Pull request merged**